libraries/rush-lib/src/logic/operations/CacheableOperationPlugin.ts

```diff
@@ -591,17 +594,25 @@ export class CacheableOperationPlugin implements IPhasedCommandPlugin {
         return;
       }
 
-      const { operationSettings, projectChangeAnalyzer } = buildCacheContext;
+      const { operationSettings, projectChangeAnalyzer, shard } = buildCacheContext;
       if (!operationSettings || !buildCacheConfiguration) {
         // Unreachable, since this will have set `cacheDisabledReason`.
         return;
       }
 
-      const projectOutputFolderNames: ReadonlyArray<string> = operationSettings.outputFolderNames || [];
+      const projectOutputFolderNames: ReadonlyArray<string> = [
+        ...(operationSettings.outputFolderNames || [])
+      ];
```

Comment:
To facilitate this being customized by plugins, I'd prefer to forward this as a property on `Operation` or `IOperationRunner`. That way the `CacheableOperationPlugin` doesn't need to understand anything about shards.

That also has the eventual benefit of allowing us to give Rush as stronger concept of `clean`, since the output folder names are what are necessary to do that.

Reply:
Do we want a flat config on the Operation or IOperationRunner? ie adding `outputFolderNames` versus `operationSettings`. I added the latter as the other properties might be useful later on?

Reply:
Fixed